From 289318a23e96a771b8d55cfa2399f094e3bbff68 Mon Sep 17 00:00:00 2001
Message-Id: <289318a23e96a771b8d55cfa2399f094e3bbff68@dist-git>
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 26 May 2017 16:05:05 +0200
Subject: [PATCH] rpc: Allow up to 256K records to be returned per domain from
 virConnectGetAllDomainStats.

The number of records that virConnectGetAllDomainStats can return per
domain is currently limited to 4096.  This is quite low -- for
example, a single guest with ~320 disks will hit this limit.  This
increases the limit to make it much larger.  Note that
VIR_NET_MESSAGE_MAX still protects the total message size in the case
where there are many domains and many disks per domain.

I tested this using a guest with 500 disks with no issues.

Signed-off-by: Richard W.M. Jones <rjones@redhat.com>
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1440683
(cherry picked from commit 4c70a6f8745714d59d4d66475f639873a0c8e191)
Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/remote/remote_protocol.x | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/remote/remote_protocol.x b/src/remote/remote_protocol.x
index 87b2bd365..3cd9604e9 100644
--- a/src/remote/remote_protocol.x
+++ b/src/remote/remote_protocol.x
@@ -233,7 +233,7 @@ const REMOTE_DOMAIN_FSFREEZE_MOUNTPOINTS_MAX = 256;
 const REMOTE_NETWORK_DHCP_LEASES_MAX = 65536;
 
 /* Upper limit on count of parameters returned via bulk stats API */
-const REMOTE_CONNECT_GET_ALL_DOMAIN_STATS_MAX = 4096;
+const REMOTE_CONNECT_GET_ALL_DOMAIN_STATS_MAX = 262144;
 
 /* Upper limit of message size for tunable event. */
 const REMOTE_DOMAIN_EVENT_TUNABLE_MAX = 2048;
-- 
2.13.0

